src/optimize/bundles_route/bundles_route.js

```diff
@@ -0,0 +1,66 @@
+import { isAbsolute, extname } from 'path';
+
+import LruCache from 'lru-cache';
+
+import { createDynamicAssetResponse } from './dynamic_asset_response';
+
+/**
+ *  Creates a route that serves files from `bundlesPath`. If the
+ *  file is js or css then it is searched for instances of
+ *  PUBLIC_PATH_PLACEHOLDER and replaces them with `publicPath`.
+ *  @param {Object} options
+ *  @property {string} options.bundlesPath
+ *  @property {string} options.basePublicPath
+ *  @return {Hapi.RouteConfig}
+ */
+export function createBundlesRoute({ bundlesPath, basePublicPath }) {
+
+  // rather than calculate the fileHash on every request, we
+  // provide a cache object to `createDynamicAssetResponse()` that
+  // will store the 100 most recently used hashes.
+  const fileHashCache = new LruCache(100);
+
+  if (typeof bundlesPath !== 'string' || !isAbsolute(bundlesPath)) {
```

Comment:
Yeah, I can include the bundlesPath value in the error message, but I figure the only time this will be triggered is when the user is monkeying around with the config file, so the more expressive error message will be more useful than seeing their value (and makes the error message longer, wrap, etc.)